bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/discovery/BusGatewayUpnpDiscovery.java

```diff
@@ -0,0 +1,229 @@
+        private String modelDescription = "<unknown>";
+        private String modelNumber = "<unknown>";
+        private String serialNumber = "<unknown>";
+        @Nullable
+        private String host;
+        private String manufacturer = "<unknown>";
+        @Nullable
+        private UDN udn;
+        private boolean isBTicino = false;
+
+        private DeviceInfo(RemoteDevice device) {
+            String deviceLog = "Discovererd device:\n+=== UPnP =========================================";
+            RemoteDeviceIdentity identity = device.getIdentity();
+            if (identity != null) {
+                this.udn = identity.getUdn();
+                deviceLog += "\n| ID.UDN       : " + udn;
+                if (identity.getDescriptorURL() != null) {
+                    deviceLog += "\n| ID.DESC URL  : " + identity.getDescriptorURL();
+                    this.host = identity.getDescriptorURL().getHost();
+                }
+                deviceLog += "\n| ID.MAX AGE : " + identity.getMaxAgeSeconds();
+            }
+            deviceLog += "\n| --------------";
+            DeviceDetails details = device.getDetails();
+            if (details != null) {
+                ManufacturerDetails manufacturerDetails = details.getManufacturerDetails();
+                if (manufacturerDetails != null) {
+                    this.manufacturer = manufacturerDetails.getManufacturer();
+                    deviceLog += "\n| MANUFACTURER : " + manufacturer + " (" + manufacturerDetails.getManufacturerURI()
+                            + ")";
+                    if (manufacturer.toUpperCase().contains("BTICINO")) {
+                        this.isBTicino = true;
+                    }
+                }
+                ModelDetails modelDetails = details.getModelDetails();
+                if (modelDetails != null) {
+                    // Model Name | Desc | Number (Uri)
+                    this.modelName = modelDetails.getModelName();
+                    this.modelDescription = modelDetails.getModelDescription();
+                    this.modelNumber = modelDetails.getModelNumber();
+                    deviceLog += "\n| MODEL        : " + modelName + " | " + modelDescription + " | " + modelNumber
+                            + " (" + modelDetails.getModelURI() + ")";
+                }
+                if (isBTicino) {
+                    this.friendlyName = details.getFriendlyName();
+                    deviceLog += "\n| FRIENDLY NAME: " + friendlyName;
+                    this.serialNumber = details.getSerialNumber();
+                    deviceLog += "\n| SERIAL #     : " + serialNumber;
+                    deviceLog += "\n| BASE URL     : " + details.getBaseURL();
+                    deviceLog += "\n| UPC          : " + details.getUpc();
+                }
+            }
+            deviceLog += "\n+==================================================";
+            logger.debug(deviceLog);
```

Comment:
This is quite a lot of code for logging. Can use the debugger instead?

Reply:
It's both logging and recognising the fields of the found UPnP device.  After many tests with users I have found so many variants of info published via UPnP -> I have found very useful to log at debug level which info is delivered via UPnP and show even more info if the device is a BTicino one. There is no other easy way to show such info to the user, who can always snooze it setting log level to INFO